python/fbprophet/diagnostics.py

```diff
@@ -414,36 +415,30 @@ def rolling_mean_by_h(x, h, w, name):
     -------
     Dataframe with columns horizon and name, the rolling mean of x.
     """
-    # Aggregate over h
     df = pd.DataFrame({'x': x, 'h': h})
     df2 = (
-        df.groupby('h').agg(['mean', 'count']).reset_index().sort_values('h')
+        df.groupby('h').agg(['sum', 'count']).reset_index().sort_values('h')
     )
-    xm = df2['x']['mean'].values
+    xs = df2['x']['sum'].values
     ns = df2['x']['count'].values
     hs = df2['h'].values
-
+    n_sum = 0
+    x_sum = 0
+    i = len(hs) - 1
+    j = i
     res_h = []
     res_x = []
-    # Start from the right and work backwards
-    i = len(hs) - 1
-    while i >= 0:
-        # Construct a mean of at least w samples.
-        n = int(ns[i])
-        xbar = float(xm[i])
-        j = i - 1
-        while ((n < w) and j >= 0):
-            # Include points from the previous horizon. All of them if still
-            # less than w, otherwise just enough to get to w.
-            n2 = min(w - n, ns[j])
-            xbar = xbar * (n / (n + n2)) + xm[j] * (n2 / (n + n2))
-            n += n2
+    while i >= 0 and j >= 0:
+        while (n_sum < w) and (j >= 0):
```

Comment:
There is a bug here where if we reach `j<0` before we reach `n_sum >= w`, we do not want to include the point in the res_x below. This should be easy to fix by putting a check before the `append` that n_sum >= w.